src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/EventHubProducer.cs

```diff
@@ -5,21 +5,33 @@ namespace MassTransit.EventHubIntegration
     using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
-    using Azure.Messaging.EventHubs;
-    using Azure.Messaging.EventHubs.Producer;
     using Initializers;
     using Logging;
+    using MassTransit.Middleware;
     using Transports;
 
 
     public class EventHubProducer :
+        Supervisor,
+        IAsyncDisposable,
         IEventHubProducer
     {
+        readonly ConnectHandle _connectHandle;
         readonly EventHubSendTransportContext _context;
 
-        public EventHubProducer(EventHubSendTransportContext context)
+        public EventHubProducer(EventHubSendTransportContext context, ConnectHandle connectHandle = null)
         {
             _context = context;
+            _connectHandle = connectHandle;
+
+            foreach (var handle in _context.GetAgentHandles())
```

Comment:
you don't ever add any handles, why bother implementing this if you aren't using it?

Reply:
I do, ProducerSupervisor

Reply:
https://github.com/MassTransit/MassTransit/pull/4012/files#diff-a0f199a611ef4cbf2ff8cc5febba3850b2f41fa9c8e7244fe22c559fcc4d2457R49

Reply:
But GetAgentHandles returns Array.Empty() for Azure Event Hub...

Reply:
fixed, thanks for catching